File: pinkmoe_server/model/request/key.go
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-06-22 11:13:07
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 08:53:29
 * @FilePath: /pinkmoe_server/model/request/key.go
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe   (如需用于商业用途或者二开，请联系作者捐助任意金额即可)
 * QQ:2419857357;支付宝:13135986153
 * Copyright (c) 2022 by coderzhaolu, All Rights Reserved.
 */
package request

// SearchKeyParams api分页条件查询及排序结构体
type SearchKeyParams struct {
	PageInfo
	Key    string `json:"key" form:"key" gorm:"comment:卡密内容"`       // 卡密内容
	Value  int    `json:"value" form:"value" gorm:"comment:卡密面额"`   // 卡密面额
	Num    int    `json:"num" form:"num" gorm:"comment:卡密面额"`       // 卡密面额
	Type   string `json:"type" form:"type" gorm:"comment:卡密类型"`     // 卡密类型
	Status string `json:"status" form:"status" gorm:"comment:卡密状态"` // 卡密状态
}

File: pinkmoe_admin/src/api/userCenter/key.ts
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-05-06 14:55:21
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 09:14:39
 * @FilePath: /pinkmoe_admin/src/api/userCenter/key.ts
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved. 
 */
import { http } from '@/utils/http/axios';
import { BasicResponseModel } from '@/api/common';

/**
 * @description: 获取卡密列表
 * @param params
 */
export function getKeyList(params?) {
  return http.request<BasicResponseModel>(
    {
      url: '/Admin/Key/KeyList',
      method: 'GET',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 更新卡密
 * @param params
 */
export function updateKey(params) {
  return http.request(
    {
      url: `/Admin/Key/KeyUpdate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 创建卡密
 * @param params
 */
export function createKey(params) {
  return http.request(
    {
      url: `/Admin/Key/KeyCreate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 删除卡密
 * @param params
 */
export function deleteKey(params) {
  return http.request(
    {
      url: `/Admin/Key/KeyDelete`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}
